packages/material-ui/src/utils/index.js

```diff
@@ -1,3 +1,4 @@
+export { default as arrowGenerator } from './arrowGenerator';
```

Comment:
Should we make it public? What use case do you have in mind?

Reply:
What do you think of keeping it inline in the Tooltip file?

Reply:
I suspect that we can get away with color inheritance and `currentColor`.

Reply:
I originally made it inline in the Tooltip file but then realized there was another example in the docs (`BootstrapTooltip`) that made use of the function so I wanted to place it somewhere that wouldn't necessitate duplicating code (in the example or in the library itself).  If you'd like it inline in the tooltip file should I just leave the same function in the example docs?

Reply:
I'm not quite clear what you mean by "color inheritance and `currentColor`".
I tried using `inherit` for `borderColor` but I get an error when I do so:
![image](https://user-images.githubusercontent.com/16123225/68702393-99d15c00-053d-11ea-9c3a-0c8e639c2ddb.png)
![image](https://user-images.githubusercontent.com/16123225/68702407-a2299700-053d-11ea-8844-ec34964748f8.png)

even after assigning borderColor to the parent `tooltip` component.

Reply:
I think that we should

1. inline all the styles in the Tooltip
2. update the demo to leverage the power of CSS

```diff
diff --git a/docs/src/pages/components/tooltips/CustomizedTooltips.tsx b/docs/src/pages/components/tooltips/CustomizedTooltips.tsx
index ac4eb8b31..5d6a5b918 100644
--- a/docs/src/pages/components/tooltips/CustomizedTooltips.tsx
+++ b/docs/src/pages/components/tooltips/CustomizedTooltips.tsx
@@ -3,7 +3,6 @@ import { withStyles, Theme, makeStyles, createStyles } from '@material-ui/core/s
 import Button from '@material-ui/core/Button';
 import Tooltip, { TooltipProps } from '@material-ui/core/Tooltip';
 import Typography from '@material-ui/core/Typography';
-import { arrowGenerator } from '@material-ui/core/utils';

 const LightTooltip = withStyles((theme: Theme) => ({
   tooltip: {
@@ -17,20 +16,9 @@ const LightTooltip = withStyles((theme: Theme) => ({
 const useStylesBootstrap = makeStyles((theme: Theme) =>
   createStyles({
     arrow: {
-      position: 'absolute',
-      fontSize: 6,
-      '&::before': {
-        content: '""',
-        margin: 'auto',
-        display: 'block',
-        width: 0,
-        height: 0,
-        borderStyle: 'solid',
-      },
+      color: theme.palette.common.black,
     },
-    popper: arrowGenerator(theme.palette.common.black),
     tooltip: {
-      position: 'relative',
       backgroundColor: theme.palette.common.black,
     },
     tooltipPlacementLeft: {
@@ -49,31 +37,9 @@ const useStylesBootstrap = makeStyles((theme: Theme) =>
 );

 function BootstrapTooltip(props: TooltipProps) {
-  const { arrow, ...classes } = useStylesBootstrap();
-  const [arrowRef, setArrowRef] = React.useState<HTMLSpanElement | null>(null);
+  const classes = useStylesBootstrap();

-  return (
-    <Tooltip
-      classes={classes}
-      PopperProps={{
-        popperOptions: {
-          modifiers: {
-            arrow: {
-              enabled: Boolean(arrowRef),
-              element: arrowRef,
-            },
-          },
-        },
-      }}
-      {...props}
-      title={
-        <React.Fragment>
-          {props.title}
-          <span className={arrow} ref={setArrowRef} />
-        </React.Fragment>
-      }
-    />
-  );
+  return <Tooltip classes={classes} arrow {...props} />;
 }

 const HtmlTooltip = withStyles((theme: Theme) => ({
diff --git a/packages/material-ui/src/Tooltip/Tooltip.js b/packages/material-ui/src/Tooltip/Tooltip.js
index d973975d4..9b09dd6b5 100644
--- a/packages/material-ui/src/Tooltip/Tooltip.js
+++ b/packages/material-ui/src/Tooltip/Tooltip.js
@@ -30,7 +30,10 @@ export const styles = theme => ({
     pointerEvents: 'auto',
   },
   /* Styles applied to the Popper component if `arrow={true}`. */
-  popperArrow: arrowGenerator(theme.palette.grey[700]),
+  popperArrow: {
+    color: theme.palette.grey[700],
+    ...arrowGenerator(),
+  },
   /* Styles applied to the tooltip (label wrapper) element. */
   tooltip: {
     backgroundColor: fade(theme.palette.grey[700], 0.9),
diff --git a/packages/material-ui/src/utils/arrowGenerator.js b/packages/material-ui/src/utils/arrowGenerator.js
index d838fdbbb..b8e0ae288 100644
--- a/packages/material-ui/src/utils/arrowGenerator.js
+++ b/packages/material-ui/src/utils/arrowGenerator.js
@@ -1,4 +1,4 @@
-export default function arrowGenerator(color) {
+export default function arrowGenerator() {
   return {
     '&[x-placement*="bottom"] $arrow': {
       top: 0,
@@ -8,7 +8,7 @@ export default function arrowGenerator(color) {
       height: '1em',
       '&::before': {
         borderWidth: '0 1em 1em 1em',
-        borderColor: `transparent transparent ${color} transparent`,
+        borderColor: `transparent transparent currentColor transparent`,
       },
     },
     '&[x-placement*="top"] $arrow': {
@@ -19,7 +19,7 @@ export default function arrowGenerator(color) {
       height: '1em',
       '&::before': {
         borderWidth: '1em 1em 0 1em',
-        borderColor: `${color} transparent transparent transparent`,
+        borderColor: `currentColor transparent transparent transparent`,
       },
     },
     '&[x-placement*="right"] $arrow': {
@@ -29,7 +29,7 @@ export default function arrowGenerator(color) {
       width: '1em',
       '&::before': {
         borderWidth: '1em 1em 1em 0',
-        borderColor: `transparent ${color} transparent transparent`,
+        borderColor: `transparent currentColor transparent transparent`,
       },
     },
     '&[x-placement*="left"] $arrow': {
@@ -39,7 +39,7 @@ export default function arrowGenerator(color) {
       width: '1em',
       '&::before': {
         borderWidth: '1em 0 1em 1em',
-        borderColor: `transparent transparent transparent ${color}`,
+        borderColor: `transparent transparent transparent currentColor`,
       },
     },
   };
```

Reply:
ahh, I'd never used `currentColor` before but now I see what you mean.
The problem is arrow `span` uses the `currentColor` of the tooltip div, not the color of the popper.  So the arrow appears the same color as the text, which is not what we want.
![image](https://user-images.githubusercontent.com/16123225/68708338-5defc400-0548-11ea-8db7-58189bb37949.png)

Reply:
Yeah, the current color only solves the override complexity part, we still need to provide the color.

Reply:
fixed in latest commit